silo-core/contracts/utils/liquidationHelper/LiquidationHelper.sol

```diff
@@ -0,0 +1,144 @@
+        DexSwapInput[] calldata _swapsInputs0x
+    ) external returns (uint256 withdrawCollateral, uint256 repayDebtAssets) {
+        require(_maxDebtToCover != 0, NoDebtToCover());
+
+        _flashLoanFrom.flashLoan(this, _debtAsset, _maxDebtToCover, abi.encode(_liquidation, _swapsInputs0x));
+        IERC20(_debtAsset).approve(address(_flashLoanFrom), 0);
+
+        withdrawCollateral = _withdrawCollateral;
+        repayDebtAssets = _repayDebtAssets;
+    }
+
+    /// @inheritdoc IERC3156FlashBorrower
+    function onFlashLoan(
+        address /* _initiator */,
+        address _debtAsset,
+        uint256 _maxDebtToCover,
+        uint256 _fee,
+        bytes calldata _data
+    )
+        external
+        returns (bytes32)
+    {
+        (
+            LiquidationData memory _liquidation,
+            DexSwapInput[] memory _swapInputs
+        ) = abi.decode(_data, (LiquidationData, DexSwapInput[]));
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), _maxDebtToCover);
+
+        bool receiveSToken;
+
+        (
+            _withdrawCollateral, _repayDebtAssets
+        ) = _liquidation.hook.liquidationCall(
+            _liquidation.collateralAsset, _debtAsset, _liquidation.user, _maxDebtToCover, receiveSToken
+        );
+
+        IERC20(_debtAsset).approve(address(_liquidation.hook), 0);
+
+        if (_liquidation.collateralAsset == _debtAsset) {
+            uint256 balance = IERC20(_liquidation.collateralAsset).balanceOf(address(this));
+            // bad debt is not supported, we will get underflow on bad debt
+            _transferToReceiver(_liquidation.collateralAsset, balance - (_maxDebtToCover + _fee));
+        } else {
+            uint256 debtBalance = IERC20(_debtAsset).balanceOf(address(this));
+            uint256 loanWithFee = _maxDebtToCover + _fee;
+
+            if (loanWithFee < debtBalance) {
+                unchecked {
+                    // safe because of `if (loanWithFee < debtBalance)`
+                    _transferToReceiver(_debtAsset, debtBalance - loanWithFee);
+                }
+            }
+
+            _executeSwap(_swapInputs); // part of collateral => _maxDebtToCover + _fee
+            uint256 balance = IERC20(_liquidation.collateralAsset).balanceOf(address(this));
+            _transferToReceiver(_liquidation.collateralAsset, balance);
+        }
+
+        IERC20(_debtAsset).approve(msg.sender, _maxDebtToCover + _fee);
+
+        return _FLASHLOAN_CALLBACK;
+    }
+
+    function _executeSwap(DexSwapInput[] memory _swapInputs) internal {
+        for (uint256 i; i < _swapInputs.length; i++) {
+            fillQuote(_swapInputs[i].sellToken, _swapInputs[i].allowanceTarget, _swapInputs[i].swapCallData);
+        }
+    }
+
+    function _transferToReceiver(address _asset, uint256 _amount) internal {
+        if (_amount == 0) return;
+
+        if (_asset == NATIVE_TOKEN) _transferNative(_amount);
+        else IERC20(_asset).transfer(TOKENS_RECEIVER, _amount);
```

Comment:
Maybe with curly braces? Or at least like this:
```suggestion
        if (_asset == NATIVE_TOKEN)
            _transferNative(_amount);
        else
            IERC20(_asset).transfer(TOKENS_RECEIVER, _amount);
```

Reply:
done